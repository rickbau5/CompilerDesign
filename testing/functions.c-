simple() {}
int functions() {}

single(int x, y, z) {}
double(int x, y, z; bool a, b, c) {}

acrossLines(int a;
            bool c;
            char a,
            d, e, f)
{
}

noDeclarations() {
    x = z;
}

noStatements() {
    int x;
    bool z;
}

calling() {
    test();
    testTwo(x, y, z);
}

arrays(int a[], b; bool c, d[]; char e[]) {}
