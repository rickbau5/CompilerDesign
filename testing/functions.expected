Func simple returns type void [line: 1]
!   Child: 0  Compound [line: 1]
Sibling: 0  Func functions returns type int [line: 2]
!   Child: 0  Compound [line: 2]
Sibling: 1  Func single returns type void [line: 4]
!   Child: 0  Param x of type int [line: 4]
!   Sibling: 0  Param y of type int [line: 4]
!   Sibling: 1  Param z of type int [line: 4]
!   Child: 1  Compound [line: 4]
Sibling: 2  Func double returns type void [line: 5]
!   Child: 0  Param x of type int [line: 5]
!   Sibling: 0  Param y of type int [line: 5]
!   Sibling: 1  Param z of type int [line: 5]
!   Sibling: 2  Param a of type bool [line: 5]
!   Sibling: 3  Param b of type bool [line: 5]
!   Sibling: 4  Param c of type bool [line: 5]
!   Child: 1  Compound [line: 5]
Sibling: 3  Func acrossLines returns type void [line: 7]
!   Child: 0  Param a of type int [line: 7]
!   Sibling: 0  Param c of type bool [line: 8]
!   Sibling: 1  Param a of type char [line: 9]
!   Sibling: 2  Param d of type char [line: 10]
!   Sibling: 3  Param e of type char [line: 10]
!   Sibling: 4  Param f of type char [line: 10]
!   Child: 1  Compound [line: 11]
Sibling: 4  Func noDeclarations returns type void [line: 14]
!   Child: 0  Compound [line: 14]
!   !   Child: 0  Assign: = [line: 15]
!   !   !   Child: 0  Id: x [line: 15]
!   !   !   Child: 1  Id: z [line: 15]
Sibling: 5  Func noStatements returns type void [line: 18]
!   Child: 0  Compound [line: 18]
!   !   Child: 0  Var x of type int [line: 19]
!   !   Sibling: 0  Var z of type bool [line: 20]
Sibling: 6  Func calling returns type void [line: 23]
!   Child: 0  Compound [line: 23]
!   !   Child: 0  Call: test [line: 24]
!   !   Sibling: 0  Call: testTwo [line: 25]
!   !   !   Child: 0  Id: x [line: 25]
!   !   !   Sibling: 0  Id: y [line: 25]
!   !   !   Sibling: 1  Id: z [line: 25]
Sibling: 7  Func arrays returns type void [line: 28]
!   Child: 0  Param a is array of type int [line: 28]
!   Sibling: 0  Param b of type int [line: 28]
!   Sibling: 1  Param c of type bool [line: 28]
!   Sibling: 2  Param d is array of type bool [line: 28]
!   Sibling: 3  Param e is array of type char [line: 28]
!   Child: 1  Compound [line: 28]
